### PR TITLE
STY Recommendations UX fixes

### LINKED ISSUE #7300: add padding to Sty Recommendations promo headline
**Is your feature request related to a problem? Please describe.**
We need to add padding to the recommendation promo headline in the story page. The values on the design are not consistent with GEL values and the advice from UX is to ensure we have:
- `8px (0.5rem) at the top, left, right and bottom.` for breakpoints < than `600px`
- `16px (1rem) at the top, left, right and bottom.` for breakpoints > than `600px`

The Story promo wrapper has a `grid-column-gap` of 1rem (>600px), 0.5(<600px) [specified here](https://github.com/bbc/psammead/blob/latest/packages/components/psammead-story-promo/src/index.jsx#L59-L63), this affords us the necessary spacing we need on the right (for ltr services) and left (for rtl services). We just need to make sure the corresponding values are applied to other sides of the promo headline.

At the moment, the story promo headline has no padding rules set and doing this might potentially impact all other areas where it's been used.

**Describe the solution you'd like**

**Describe alternatives you've considered**
1. since this is only for recommendations, allow the [story promo headline](https://github.com/bbc/psammead/blob/latest/packages/components/psammead-story-promo/src/index.jsx#L125-L146) accept an `isRecommendation` prop to enable this  set of padding rules and ensure the values are consistent between ltr and rtl services. 
2. create a styled headline for the story promo usage [here](https://github.com/bbc/simorgh/blob/latest/src/app/containers/StoryPromo/index.jsx#L162), adding the padding rules thereby overriding it's default behavior.

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [X] This feature is expected to need manual testing.
- Ensure there is no change to the story promo usages on the front and idx pages

**Additional context**
- rtl and ltr local sty pages http://localhost:7080/arabic/23298105, http://localhost:7080/mundo/23263889

### PR DESCRIPTION
**Pull request by HarveyPeachey**
Resolves #7300

**Overall change:**
Adds spacing for headline text in each recommendation list item. Added missing `ConstrainedGrid` to keep with consistency of other blocks in the main column. And added spacing on the bottom of the recommendations container.

**Code changes:**

- Added a `StyledHeadline` in `StoryPromo` which applies padding to the headline if `isRecommendations` is set to `true` and varies based on services direction.
- Added `<GridItemConstrainedMediumNoMargin>` for layout following on from other blocks in the main column.
- Added bottom margin for spacing out recommendations container for other blocks.
- Updated snapshots and added rtl snapshot

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams - Added cross-team label due to snapshot changes on other pages.
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

**Review by chris-hinds — APPROVED** [review]
Checked out and ran locally, spacing looks good to me 👍

File: src/app/containers/StoryPromo/index.jsx [hunk 1]
```diff
@@ -156,15 +159,28 @@ const StoryPromoContainer = ({
   const headingTagOverride =
     isRecommendation || isContentTypeGuide ? 'div' : null;
 
+  const StyledHeadline = styled(Headline)`
+    ${({ recommendationSpacing }) =>
+      recommendationSpacing &&
+      `
+      padding: ${GEL_SPACING} ${GEL_SPACING} ${GEL_SPACING} 0;
```
Comment (Simon2828) [inline src/app/containers/StoryPromo/index.jsx]: Does this work ok for RTL pages too?

Comment (HarveyPeachey) [reply src/app/containers/StoryPromo/index.jsx]: Good spot, it doesn't. Will amend 👍

File: src/app/containers/CpsRecommendations/index.jsx [hunk 2]
```diff
@@ -119,28 +125,30 @@ const CpsRecommendations = ({ items, parentColumns }) => {
...
-          singleTransform={singleTransform}
-          listTransform={listTransform}
-          sectionLabelOverrideAs="strong"
-          sectionLabelBar={false}
-          sectionLabelBackground={C_LUNAR}
-          columnType="main"
-        />
-      </RecommendationsWrapper>
-    </SkipLinkWrapper>
+    <GridItemConstrainedMediumNoMargin>
```
Comment (rhenshaw56) [inline src/app/containers/CpsRecommendations/index.jsx:128]: ❤️

**Review by rhenshaw56 — APPROVED** [review]

**Review by Simon2828 — APPROVED** [review]
Looks really good!

**Comment by PriyaKR** [comment]
Looks good to me.

**Pull request merged**